### PR TITLE
Add view change recovery test

### PR DESCRIPTION
**Pull request by StefanPalashev**
Add a test that checks the recovery of fast path of replicas after they have
initiated a view change.

### CODE REVIEW

File: tests/apollo/test_skvbc_view_change.py [hunk 1]
```diff
@@ -452,6 +452,52 @@ async def write_req():
...
                          "Make sure we are still on the initial view.")
 
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: c == 0, rotate_keys=True)
+    @verify_linearizability()
+    async def test_recovering_of_replicas_with_initiated_view_change(self, bft_network, tracker):
+        """
+        The goal of this test is to validate recovery to fast path after stopping all replicas during view change.
+        Test is valid only when c = 0.
+        To trigger view change we start N - (F+1) replicas. This causes fast view change because 3f+2*0+1-f-1=2f
```
Comment (HristoStaykov) [inline tests/apollo/test_skvbc_view_change.py:462]: "This causes fast view change" -> "This causes View Change to be initiated, but will not be completed"

**Review by HristoStaykov — APPROVED** [review]
LGTM, thanks, @StefanPalashev ! Just a minor description correction.

File: tests/apollo/util/bft.py [hunk 2]
```diff
@@ -1321,4 +1321,29 @@ async def the_replica_to_ask_for_view_change():
...
+                    for replica_id in replicas_ids:
+                        nursery.start_soon(self.wait_for_replica_to_reach_view, replica_id, expected_view)
+            
+    async def wait_for_replica_to_reach_view(self, replica_id, expected_view):
+        """
+        Wait for a single replica to reach the expected view.
+        """
+        with log.start_action(action_type="wait_for_replica_to_reach_view", replica=replica_id, expected_view=expected_view) as action:
+            async def expected_view_to_be_reached():
+                key = ['replica', 'Gauges', 'view']
```
Comment (teoparvanov) [inline tests/apollo/util/bft.py:1342]: Are we sure this metric is incremented when the replica does `MoveToHigherView`?

**Review by teoparvanov — APPROVED** [review]
LGTM, thanks!

File: tests/apollo/util/bft.py (hunk 2 above)
Comment (StefanPalashev) [reply tests/apollo/util/bft.py:1342]: Yes, this is the correct metric. The following LOC (Currently 2431) in `MoveToHigherView`'s body sets the value of the metric:
`metric_view_.Get().Set(nextView);`

File: tests/apollo/test_skvbc_view_change.py (hunk 1 above)
Comment (StefanPalashev) [reply tests/apollo/test_skvbc_view_change.py]: The description has been corrected. Thanks!

**Pull request merged**